### PR TITLE
implemented MachineServiceImpl method realization with parametrization and wildcards

### PR DESCRIPTION
**Pull request by CodeWebWeaver**

### CODE REVIEW

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 1]
```diff
@@ -1,38 +1,13 @@
...
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import java.util.List;
+
+public interface MachineService<T extends Machine> {
+
+    List<T> getAll(Class<? extends T> type);
+
+    <T extends Machine> void fill(List<? super T> list, T value);
```
Comment (DmytroV95) [inline src/main/java/core/mate/academy/service/MachineService.java:10]: We need just change Object type where it specified  to other type
```suggestion
    void fill(List<? super T> list, T value);

```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 2]
```diff
@@ -1,7 +1,40 @@
...
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Your implementation of MachineService.
+ */
+public class MachineServiceImpl implements MachineService<Machine> {
+
```
Comment (DmytroV95) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:15]: delete redunndant line

**Review by DmytroV95 — CHANGES_REQUESTED** [review]

File: src/main/java/core/mate/academy/service/MachineService.java (hunk 1 above)
Comment (Skarbey) [reply src/main/java/core/mate/academy/service/MachineService.java]: Oookay

**Review by DmytroV95 — APPROVED** [review]
Good job!

**Review by irynamekh — APPROVED** [review]

File: src/main/java/core/mate/academy/model/Bulldozer.java [hunk 3]
```diff
@@ -1,13 +1,41 @@
...
+
     public Bulldozer() {
     }
 
+    public Bulldozer(int bulldozerId, String model) {
+        this.bulldozerId = bulldozerId;
+        this.model = model;
+    }
+
+    public int getBulldozerId() {
```
Comment (andrii-hoienko) [inline src/main/java/core/mate/academy/model/Bulldozer.java:15]: dont create methods you don`t use

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Machine> {
```
Comment (andrii-hoienko) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:7]: ExcavatorProducer returns Machine? Or maybe he should return some specific machine

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 5]
```diff
@@ -1,7 +1,35 @@
...
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        List<T> machineList = new ArrayList<>();
+        if (type.equals(Truck.class)) {
+            machineList = (List<T>) new ArrayList<>(new TruckProducer().get());
```
Comment (andrii-hoienko) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:15]: please think how to avoid casting

Comment (andrii-hoienko) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:15]: new TruckProducer() should be class level variable

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good, but let`s improve your solution

**Pull request closed without merging**